GPDB DOCS - postGIS extension

mkiyama opened the pull request:
Updated documentation for the GPDB postGIS extension.

NOTE: incomplete information about creating postGIS extension for GPDB OSS.



[ci skip]

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -36,52 +33,63 @@
-        <p>The Greenplum Database PostGIS extension is available from <xref
-            href="https://network.pivotal.io" scope="external" format="html"><ph>Pivotal
-              Network</ph></xref>. You can install it using the Greenplum Package Manager
+        <!--For OSS, compile and install the PostGIS extension package. -->
```
Per latest info from Frank, we can get rid of OSS instructions for PostGIS.  I've already conditionalized the map file so postGIS.xml isn't currently getting built in.

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -96,48 +104,70 @@
+          <li><codeph>raster_comments.sql</codeph> - Add comments to the PostGIS Raster
+            functions.</li>
+        </ul></p>
+      <p>You can view comments with the <codeph>pslq</codeph> meta-command <codeph>\dd
```
pslq -> psql

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -96,48 +104,70 @@
+            <varname>function_name</varname></codeph> or from any tool that can show Greenplum
+        Database function comments.</p>
+      <p>For example, these commands run the SQL scripts <codeph>postgis.sql</codeph>,
+          <codeph>rtpostgis.sql</codeph> and <codeph>spatial_ref_sys.sql</codeph> in the database
```
Add comma after rtpostgis.sql

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -96,48 +104,70 @@
+        <codeblock>export GDAL_DATA=$GPHOME/share/gdal
+export POSTGIS_ENABLE_OUTDB_RASTERS=0
+export POSTGIS_GDAL_ENABLED_DRIVERS=ENABLE_ALL</codeblock>
+        <p><codeph>GDAL_DATA</codeph> specifies location of and GDAL utilities and support files
```
Remove the first "and" from this sentence.

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -96,48 +104,70 @@
+export POSTGIS_GDAL_ENABLED_DRIVERS=ENABLE_ALL</codeblock>
+        <p><codeph>GDAL_DATA</codeph> specifies location of and GDAL utilities and support files
+          used by the GDAL library. For example, in order for GDAL to properly evaluate EPSG codes
+          requires EPSG support files (so called dictionaries, mostly in ​CSV format), such as
```
Something missing from this sentence.  I think just adding "**Greenplum Database** requires" fixes it.

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -96,48 +104,70 @@
+            <codeph>GDAL_DATA</codeph> directory.</p>
+        <p><codeph>POSTGIS_GDAL_ENABLED_DRIVERS</codeph> sets the enabled GDAL drivers in the
+          PostGIS environment.</p>
+        <p><codeph>POSTGIS_ENABLE_OUTDB_RASTERS</codeph> is boolean configuration option to enable
```
is **a** boolean

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -96,48 +104,70 @@
   </topic>
   <topic id="topic6" xml:lang="en">
-    <title id="ij169830">Migrating from PostGIS 1.4 to 2.0</title>
+    <title id="ij169830">Migrating from PostGIS 2.0 to 2.1</title>
```
Can we genericize this entire topic?  5.0.0 users will need to start with 2.1, so really there's no immediate migration scenario to document.  I'd rather have this talk about <older-version. and <newer-version> generically.

mkiyama replied:
I added this topic if somebody was brave enough to import a 4.3.x database w/postgis to 5.0.
Apparently, it is technically possible.

I'll remove this topic for now, until upgrade path becomes clearer.

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -231,16 +285,14 @@ USING GIST ( <i>geometryfield</i> );</codeblock>
+          <li>Data types and functions related to PostGIS topology functionality, such as
+              <apiname>TopoGeometry</apiname>, are not supported by Greenplum Database.</li>
+          <li> Functions that perform <codeph>ANALYZE</codeph> operations for user-defined data
+            types are not supported. For example, <apiname>ST_Estimated_Extent</apiname> function is
```
**the** ST_Estimated_Extent function

dyozie commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -36,52 +33,63 @@
         and processing of GIS objects. </p>
-      <p>Go to<xref href="http://postgis.refractions.net/" scope="external" format="html">
-          http://postgis.refractions.net/</xref> for more information about PostGIS.</p>
+      <p>The Greenplum Database PostGIS extension supports the PostGIS extension PostGIS Raster. The
```
This is the only place that Raster is called out as an PostGIS extension, and it's confusing to talk about two different types of "extension" in the same paragraph.  I think it would be clearer to just say:  

The Greenplum Database PostGIS extension supports the PostGIS raster data type and most PostGIS Raster functions. For example, the PostGIS geometry data type offerss a single set..

mkiyama commented:
@dyozie 
I updated the PostGIS doc based on your review comments.
I also added information order of installing and uninstalling PostGIS and PostGIS Raster.

mkiyama commented:
@orhankislal @HaozhouWang 
Can you please review the PostGIS doc?
thanks

litzell commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -96,48 +98,92 @@
+      <p>To remove PostGIS support from a database, run SQL scripts that are supplied with the
+        PostGIS extension package in <codeph>$GPHOME/share/postgresql/contrib/postgis-2.1/</codeph>
+      </p>
+      <note>If you installed, PostGIS Raster, you must uninstall the PostGIS Raster before you
```
delete comma after installed

litzell commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -157,8 +203,8 @@ INSERT INTO geom_test ( gid, geom, name )
 SELECT * from geom_test WHERE geom &amp;&amp;
   Box3D(ST_GeomFromEWKT('LINESTRING(2 2 0, 3 3 0)'));</codeblock>
       <p>The following example SQL statements create a table, adds a geometry column to the table
```
adds -> add

litzell commented on gpdb-doc/dita/ref_guide/extensions/postGIS.xml:
```diff
@@ -157,8 +203,8 @@ INSERT INTO geom_test ( gid, geom, name )
-        with a SRID integer value that references an entry in the SPATIAL_REF_SYS table. The
-          <codeph>INSERT</codeph> statements add to geopoints to the table.</p>
+        with a SRID integer value that references an entry in the <codeph>SPATIAL_REF_SYS</codeph>
+        table. The <codeph>INSERT</codeph> statements add to geopoints to the table.</p>
```
"to geopoints" or "two geopoints"?

orhankislal commented:
Looks accurate to me.
